### PR TITLE
feat(configurable-thresholds): User project thresholds for Apdex and User Misery in Performance

### PR DESCRIPTION
**Pull request by shruthilayaj**
The backend components have been added to support project
level transaction thresholds for Apdex and User Misery. This will 
now use the new Apdex and User Misery in the Performance page.

### CODE REVIEW

File: static/app/utils/discover/fieldRenderers.tsx [hunk 1]
```diff
@@ -435,11 +435,17 @@ const SPECIAL_FUNCTIONS: SpecialFunctions = {
...
     let userMiseryField: string = '';
     let countMiserableUserField: string = '';
+    let projectThresholdConfig: string = '';
     for (const field in data) {
       if (field.startsWith('user_misery')) {
         userMiseryField = field;
-      } else if (field.startsWith('count_miserable_user')) {
+      } else if (
+        field.startsWith('count_miserable_user') ||
+        field.startsWith('count_miserable_new_user')
```
Comment (k-fish) [inline static/app/utils/discover/fieldRenderers.tsx:444]: ~~Where does `count_miserable_new_user` come from?~~ nvm see the field below.

File: static/app/utils/discover/fieldRenderers.tsx [hunk 2]
```diff
@@ -450,7 +456,10 @@ const SPECIAL_FUNCTIONS: SpecialFunctions = {
     const uniqueUsers = data.count_unique_user;
     const userMisery = data[userMiseryField];
 
-    const miseryLimit = parseInt(userMiseryField.split('_').pop() || '', 10) || undefined;
+    let miseryLimit = parseInt(userMiseryField.split('_').pop() || '', 10);
+    if (isNaN(miseryLimit)) {
```
Comment (k-fish) [inline static/app/utils/discover/fieldRenderers.tsx:460]: What does userMiseryField become that this `NaN` check succeeds and we switch to using projectThresholdConfig below?

File: tests/js/spec/views/performance/data.spec.jsx [hunk 3]
```diff
@@ -90,4 +90,72 @@ describe('generatePerformanceEventView()', function () {
...
+      expect.arrayContaining([
+        expect.objectContaining({field: 'count_miserable_new(user)'}),
+      ])
+    );
+    expect(result.fields).not.toEqual(
+      expect.arrayContaining([expect.objectContaining({field: 'apdex_new()'})])
+    );
+
+    const newOrganization = TestStubs.Organization({
+      apdexThreshold: 300,
```
Comment (k-fish) [inline tests/js/spec/views/performance/data.spec.jsx:125]: Is apdex also 300 by default? I'd maybe change this to some other number to ensure the default isn't still being used? Minor though.

**Review by k-fish — APPROVED** [review]
👍

File: static/app/utils/discover/fieldRenderers.tsx (hunk 2 above)
Comment (shruthilayaj) [reply static/app/utils/discover/fieldRenderers.tsx:460]: Good question, `userMiseryField` can either be `user_misery(X00)` or `user_misery_new()` which will use the threshold values configured on the project level, so miseryLimit may be NaN

**Pull request merged**